include/rapidcheck/state/Commands.hpp

```diff
@@ -49,5 +196,81 @@ void showValue(const std::vector<
   }
 }
 
+template <typename CommandType>
+struct ParallelCommands {
+  using Cmd = CommandType;
+  using Cmds = Commands<CommandType>;
+
+  ParallelCommands() {}
+
+  ParallelCommands(const Cmds &prefix, const Cmds &left, const Cmds &right)
+      : prefix(prefix)
+      , left(left)
+      , right(right) {}
+
+  Cmds prefix;
+  Cmds left;
+  Cmds right;
+};
+
+template <typename Cmd, typename Model, typename Sut>
+void runAllParallel(const ParallelCommands<Cmd> &commands, const Model &state, Sut &sut) {
+
+  // Verify pre-conditions for all possible interleavings
+  detail::isValidSequence(commands, state);
```

Comment:
`isValidSequence` returns `void` which makes it weird that it is prefixed with `is`. If the expectation is that it should throw, it should probably be prefixed with `verify` of `assert` instead.